SubEthaEdit-Mac/Source/PlainTextDocument.m

```diff
@@ -3549,21 +3576,33 @@ - (BOOL)writeToURL:(NSURL *)absoluteURL ofType:(NSString *)inType forSaveOperati
 	//-timelog    NSDate *startDate = [NSDate date];
 	//-timelog    NSLog(@"%s %@ %@ %d %@",__FUNCTION__, absoluteURL, inTypeName, saveOperation,originalContentsURL);
     DEBUGLOG(@"FileIOLogDomain", AllLogLevel, @"write to:%@ type:%@ saveOperation:%lu originalURL:%@", absoluteURL, inType, (unsigned long)saveOperation,originalContentsURL);
+    
+    NSData *writtenData;
+    
     if (UTTypeConformsTo((__bridge CFStringRef)inType, kUTTypeData)) {
         BOOL modeWantsUTF8BOM = [[[self documentMode] defaultForKey:DocumentModeUTF8BOMPreferenceKey] boolValue];
         DEBUGLOG(@"FileIOLogDomain", SimpleLogLevel, @"modeWantsUTF8BOM: %d, hasUTF8BOM: %d", modeWantsUTF8BOM, I_flags.hasUTF8BOM);
         BOOL useUTF8Encoding = ((I_lastSaveOperation == NSSaveToOperation) && (I_encodingFromLastRunSaveToOperation == NSUTF8StringEncoding)) || ((I_lastSaveOperation != NSSaveToOperation) && ([self fileEncoding] == NSUTF8StringEncoding));
 		BOOL result = NO;
         if ((I_flags.hasUTF8BOM || modeWantsUTF8BOM) && useUTF8Encoding) {
             NSData *data = [[[self textStorage] string] dataUsingEncoding:NSUTF8StringEncoding allowLossyConversion:NO];
-            result = [[data dataPrefixedWithUTF8BOM] writeToURL:absoluteURL options:0 error:outError];
+            writtenData = [data dataPrefixedWithUTF8BOM];
+            result = [writtenData writeToURL:absoluteURL options:0 error:outError];
             if (result) {
             	// write the xtended attribute for utf-8
             	[UKXattrMetadataStore setString:@"UTF-8;134217984" forKey:@"com.apple.TextEncoding" atPath:[absoluteURL path] traverseLink:YES];
+            } else {
+                writtenData = nil;
             }
         } else {
             // let us write using NSStrings write methods so the encoding is added to the extended attributes
-            result = [[[(FoldableTextStorage *)[self textStorage] fullTextStorage] string] writeToURL:absoluteURL atomically:NO encoding:[self fileEncoding] error:outError];
+            NSString *text = [[(FoldableTextStorage *)[self textStorage] fullTextStorage] string];
+            NSStringEncoding encoding = [self fileEncoding];
+            
+            result = [text writeToURL:absoluteURL atomically:NO encoding:encoding error:outError];
```

Comment:
This one makes me a little bit uneasy. We only ose the string method to also write the system wide xattrs it writes. Maybe we should handle the com.apple.TextEncoding safe always on ourn own and make sure the NSData really is the thing we wrote.